frontend/src/pages/coding/codingPage.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useState } from "react";
+import Image from "next/image";
+import Topbar from "@/components/elements/Topbar";
+import { Button } from "@/components/ui/button";
+import { Textarea } from "@/components/ui/textarea";
+import PrismLoader from "@/components/prism-loader";
+import { Separator } from "@/components/ui/separator";
+import Prism from "prismjs";
+
+function CodingPage() {
+  const [isTopOfPage, setIsTopOfPage] = useState<boolean>(true);
+  const [code, setCode] = useState("");
+  const [showResult, setShowResult] = useState("");
+  const [count, setCount] = useState(0);
+  useEffect(() => {
+    const handleScroll = () => {
+      if (window.scrollY === 0) {
+        setIsTopOfPage(true);
+      }
+      if (window.scrollY !== 0) {
+        setIsTopOfPage(false);
+      }
+    };
+    window.addEventListener("scroll", handleScroll);
+    return () => window.removeEventListener("scroll", handleScroll);
+  }, []);
+  function update(text) {
+    let result_element = document.querySelector("#highlighting-content");
+    if (text[text.length - 1] == "\n") {
+      // If the last character is a newline character
+      text += " "; // Add a placeholder space character to the final line
+    }
+    // Update code
+    result_element.innerHTML = text
+      .replace(new RegExp("&", "g"), "&")
+      .replace(new RegExp("<", "g"), "<");
+    // Syntax Highlight
+    Prism.highlightElement(result_element);
+  }
+  function sync_scroll(element) {
+    /* Scroll result to scroll coords of event - sync with textarea */
+    let result_element = document.querySelector("#highlightedCodingContent");
+    // Check if result_element is not null
+    if (result_element) {
+      // Get and set x and y
+      result_element.scrollTop = element.scrollTop;
+      console.log("top: ", result_element.scrollTop);
+
+      result_element.scrollLeft = element.scrollLeft;
+      console.log("left: ", result_element.scrollLeft);
+    }
+  }
+  function check_tab(element, event) {
```

Comment:
j'ai un peu modif ta fonction, je lui e au lieu de e.taget et e. Au final tu passe 2 fois la même chose, de plus qd tu appel ta fonction vsCode me marque un vieux message: "event is deprecated"

```
	function check_tab(e) {
		const code = e.target.value;

		if (e.key == "Tab") {
			/* Tab key pressed */
			e.preventDefault(); // stop normal

			const before_tab = code.slice(0, e.target.selectionStart); // text before tab
			const after_tab = code.slice(
				e.target.selectionEnd,
				e.target.value.length
			); // text after tab
			const cursor_pos = e.target.selectionEnd + 1; // where cursor moves after tab - moving forward by 1 char to after tab

			e.target.value = before_tab + "\t" + after_tab; // add tab char

			// move cursor
			e.target.selectionStart = cursor_pos;
			e.target.selectionEnd = cursor_pos;

			update(e.target.value); // Update text to include indent
		
```